test/helpers/foundry/options.ts

```diff
@@ -0,0 +1,142 @@
+  Architecture,
+  Binary,
+  Platform,
+} from './types';
+
+export function printBanner() {
+  console.log(`
+.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
+
+ ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
+ ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development
+ ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.
+
+.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
+
+Repo       : https://github.com/foundry-rs/
+Book       : https://book.getfoundry.sh/
+Chat       : https://t.me/foundry_rs/
+Support    : https://t.me/foundry_support/
+Contribute : https://github.com/orgs/foundry-rs/projects/2/
+
+.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
+`);
+}
+
+export function parseArgs(args: string[] = argv.slice(2)) {
+  const { $0, _, ...parsed } = yargs()
+    // Ensure unrecognized commands/options are reported as errors.
+    .strict()
+    // disable yargs's version, as it doesn't make sense here
+    .version(false)
+    // use the scriptName in `--help` output
+    .scriptName('yarn foundryup')
+    // wrap output at a maximum of 120 characters or `stdout.columns`
+    .wrap(Math.min(120, stdout.columns))
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true,
+    })
+    // enable ENV parsing, which allows the user to specify foundryup options
+    // via environment variables prefixed with `FOUNDRYUP_`
+    .env('FOUNDRYUP')
+    .command(['$0', 'install'], 'Install foundry binaries', (builder) => {
+      builder.options(getOptions()).pkgConf('foundryup');
+    })
+    .command('cache', '', (builder) => {
+      builder.command('clean', 'Remove the shared cache files').demandCommand();
+    })
+    .parseSync(args);
+
+  const command = _.join(' ');
+  if (command === 'cache clean') {
+    return {
+      command,
+    } as const;
+  }
+
+  // if we get here `command` is always 'install' or '' (yargs checks it)
+  return {
+    command: 'install',
+    options: parsed as ParsedOptions<ReturnType<typeof getOptions>>,
+  } as const;
+}
+
+const Binaries = Object.values(Binary) as BinariesTuple;
+
+function getOptions(
+  defaultPlatform = platform(),
+  defaultArch = normalizeSystemArchitecture(),
+) {
+  return {
+    binaries: {
+      alias: 'b',
+      type: 'array' as const,
+      multiple: true,
+      description: 'Specify the binaries to install',
+      default: Binaries,
+      choices: Binaries,
+      coerce: (values: Binary[]): Binary[] => [...new Set(values)], // Remove duplicates
+    },
+    checksums: {
+      alias: 'c',
+      description: 'JSON object containing checksums for the binaries.',
+      coerce: (rawChecksums: string | Checksums): Checksums => {
+        try {
+          return typeof rawChecksums === 'string'
+            ? JSON.parse(rawChecksums)
+            : rawChecksums;
+        } catch {
+          throw new Error('Invalid checksums');
```

Comment:
we could do a lot more validation of the checksum object... but I think it's bit premature, since maybe the way I implemented it is horrible.